### DISCUSSION TITLE
Pagination for finding records with newly added Dao().FindRecordsByFilter ?

### DISCUSSION THREAD

**Discussion by totorogendut** [pocketbase/pocketbase#3074] [Q&A] [answered]
Is there any way to implement pagination with [Dao().FindRecordsByFilter](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.17.1/daos#Dao.FindRecordsByFilter)

There are filter, sort, and limit but there is a lack for list offset to implement pagination with it. 
Appreciate it if anyone can help me with a workaround if this `FindRecordsByFilter` function doesn't mean to be used for paginated data.

**Comment by ganigeorgiev** [comment]
None of the `Dao` builtin methods perform pagination.

The builtin pagination api (aka. the [`search` subpackage](https://pkg.go.dev/github.com/pocketbase/pocketbase/tools/search)) currently is a little more complicated as it depends on what actually you want to allow your users to being able to filter/sort (_for a basic example you can check the [`search.NewProvider` godoc](https://pkg.go.dev/github.com/pocketbase/pocketbase/tools/search#NewProvider)_).

You don't have to use the above and you can construct your own `Dao.RecordQuery()` with explicitly set limit and offset.

I can try to provide an example, but there is not enough context.
Could you elaborate a little more what is your use case and what you are trying to do?

**Reply by ganigeorgiev** [reply]
_I guess adding an `offset` argument to `Dao.FindRecordsByFilter` makes sense as usually it is used in combination with limit anyway, but that will be for v0.18.0._

**Reply by totorogendut** [reply]
I'm trying to move this into the backend because I want to implement my own custom business logic

```js
let page = 1;
let limit = 20;
const {items, totalItems, totalPages} = await db.collection("images").getList(page, limit, {
  filter: "isPublic = true",
  sort: "-created",
});
```

and in the backend currently I have a `GET` serve event with `/api/image-data/list/:page/:limit` route,
which the code goes something like

```go
limit, err := strconv.Atoi(c.PathParam("limit"))
if err != nil {
  return apis.NewBadRequestError(err.Error(), nil)
}

page, err := strconv.Atoi(c.PathParam("page"))
if err != nil {
  return apis.NewBadRequestError(err.Error(), nil)
}

// TODO - add pagination feature to image list
imageRecords, err := e.App.Dao().FindRecordsByFilter("images", "isPublic = True", "-created", limit)
```

**Reply by ganigeorgiev** [reply, marked as answer]
Since your filter doesn't require the use of nested fields lookups or anything else more complex, it might be easier to construct the query on your own with something like:

```go
records := []*models.Record{}

err := e.App.Dao().RecordQuery("images").
	AndWhere(dbx.HashExp{"isPublic": true}).
	OrderBy("created DESC").
	Limit(int64(limit)).
	Offset(int64(offset)).
	All(&records)
```

**Reply by totorogendut** [reply]
Awesome! Thanks!
